test/unit/int_algolia/scripts/algolia/helper/orderHelper.test.js

```diff
@@ -0,0 +1,134 @@
+
+// Mock the Site module
+jest.mock('dw/system/Site', () => {
+    return {
+        getCurrent: function () {
+            return {
+                getAllowedLocales: function () {
+                    var arr = ['default', 'fr', 'en'];
+                    arr.size = function () {
+                        return arr.length;
+                    };
+                    return arr;
+                }
+            };
+        }
+    }
+}, { virtual: true });
+
+// Mock AlgoliaLocalizedProduct
+jest.mock('*/cartridge/scripts/algolia/model/algoliaLocalizedProduct', () => {
+    return function(config) {
+        return {
+            objectID: config.product ? config.product.ID : 'test-product-id',
+            locale: config.locale,
+            in_stock: config.product ? config.product.in_stock : true,
+            price: {
+                USD: 129.99,
+                EUR: 119.99
+            },
+            _attributeList: config.attributeList
+        };
+    };
+}, { virtual: true });
+
+// Mock algoliaData
+jest.mock('*/cartridge/scripts/algolia/lib/algoliaData', () => {
+    return {
+        calculateIndexName: function (indexType, locale) {
+            return indexType + '_' + locale;
+        }
+    };
+}, { virtual: true });
+
+// Mock jobHelper
+jest.mock('*/cartridge/scripts/algolia/helper/jobHelper', () => {
+    return {
+        AlgoliaOperation: function (action, obj, indexName) {
+            return {
+                action: action,
+                obj: obj,
+                indexName: indexName
+            };
+        }
+    };
+}, { virtual: true });
+
+const orderHelper = require('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/helper/orderHelper');
+
+describe('Order Helper', function () {
+    test('generateAlgoliaOperations should create operations for all locales', function () {
+        const masterProduct = {
+            ID: 'test-product-id',
+            in_stock: true
+        };
+
+        const productConfig = {
+            product: masterProduct,
+            attributeList: ['variants', 'in_stock']
+        };
+
+        const operations = orderHelper.generateAlgoliaOperations(productConfig);
+
+        // Should have 3 operations (one for each locale)
+        expect(operations.length).toBe(3);
+
+        // Check the default locale operation
+        expect(operations[0].action).toBe('partialUpdateObject');
+        expect(operations[0].indexName).toBe('products_default');
+        expect(operations[0].obj.objectID).toBe('test-product-id');
+        expect(operations[0].obj._attributeList).toEqual(['variants', 'in_stock']);
+        expect(operations[0].obj.in_stock).toBe(true);
+
+        // Check the fr locale operation
+        expect(operations[1].action).toBe('partialUpdateObject');
+        expect(operations[1].indexName).toBe('products_fr');
+        expect(operations[1].obj.locale).toBe('fr');
+        expect(operations[1].obj._attributeList).toEqual(['variants', 'in_stock']);
```

Comment:
This does not test what will really happen in reality, here it actually tests your mock function (it's your mock function that sets `locale`.
Ideally you shouldn't have to mock `algoliaLocalizedProduct`, and the test should assert the final AlgoliaOperations generated by the function (it's a good use case for snapshots here)

Reply:
used snapshots instead of it.